ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala

```diff
@@ -47,18 +47,18 @@ final class ContractKeys(session: LedgerSession) extends LedgerTestSuite(session
         // fetch by key should fail during interpretation
         // Reason: Only stakeholders see the result of fetchByKey, beta is neither stakeholder nor divulgee
         fetchFailure <- beta
-          .exercise(delegate, delegation.exerciseFetchByKeyDelegated(_, owner, key, None))
+          .exercise(delegate, delegation.exerciseFetchByKeyDelegated(_, owner, key))
           .failed
 
         // lookup by key delegation is should fail during validation
         // Reason: During command interpretation, the lookup did not find anything due to privacy rules,
         // but validation determined that this result is wrong as the contract is there.
         lookupByKeyFailure <- beta
-          .exercise(delegate, delegation.exerciseLookupByKeyDelegated(_, owner, key, None))
+          .exercise(delegate, delegation.exerciseLookupByKeyDelegated(_, owner, key))
           .failed
```

Comment:
This would seem to lead to unpredictable behaviour.  If lookup is done using lookupByKey on a non-existent key and it returns None I don't see why that behaviour should change if that key is then used by a different party in a way that is not visible to the original caller ?

Reply:
Consider the following situation:
`Alice` can send a command that results in a transaction with a `NodeLookupByKey` with the right authorization, but `Alice` is not actually a stakeholder of the contract that the lookup would resolve to, therefore the `NodeLookupByKey` doesn't find a contract. Based on that fact, the DAML model can make a decision (i.e. create a contract with such a key). However, during validation the committer sees that there already exists such a contract. In that case the validation needs to fail, because the original transaction didn't have an accurate view of the ledger.

Similarly, if there is contention on the contract key, the validation shows that the original `NodeLookupByKey` and all following actions are invalid.

Reply:
I see, so there are three possible outcomes.  Visible(conctractId), Available and Unavailable.  Could we change the API to reflect this more obviously ?